### PR TITLE
Add section on Cookie-to-header token (#1215)

### PR DESCRIPTION
**Pull request by phalgunv**
Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [X] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [X] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [X] Any references to websites have been formatted as [TEXT](URL)
- [X] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #1215.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md [hunk 1]
```diff
@@ -59,6 +59,37 @@ For example:
 
 Inserting the CSRF token in a custom HTTP request header via JavaScript is considered more secure than adding the token in the hidden field form parameter because requests with custom headers are automatically subject to the same-origin policy.
 
+### Cookie-to-header token
+
+Web applications that use JavaScript for the majority of their operations may use the following anti-CSRF technique:
+
+On an initial visit without an associated server session, the web application sets a cookie. The cookie typically contains a random token which may remain the same for up to the life of the web session
```
Comment (szh) [inline cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md:66]:
```suggestion
On an initial visit without an associated server session, the web application sets a cookie. The cookie typically contains a random token which may remain the same for up to the life of the web session.
```

File: cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md [hunk 2]
```diff
@@ -59,6 +59,37 @@ For example:
...
+
+```
+Set-Cookie: __Host-csrf_token=i8XNjC4b8KVok4uw5RftR38Wgp2BFwql; Expires=Thu, 23-Jul-2015 10:25:33 GMT; Max-Age=31449600; Path=/; SameSite=Lax; Secure
+```
+
+JavaScript operating on the client side reads its value and copies it into a custom HTTP header sent with each transactional request
```
Comment (szh) [inline cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md:72]:
```suggestion
JavaScript operating on the client side reads its value and copies it into a custom HTTP header sent with each transactional request.
```

File: cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md [hunk 3]
```diff
@@ -59,6 +59,37 @@ For example:
...
+
+```
+X-Csrf-Token: i8XNjC4b8KVok4uw5RftR38Wgp2BFwql
+```
+
+The server validates presence and integrity of the token
```
Comment (szh) [inline cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md:78]:
```suggestion
The server validates presence and integrity of the token.

```

File: cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md [hunk 4]
```diff
@@ -59,6 +59,37 @@ For example:
...
+Security of this technique is based on the assumption that only JavaScript running on the client side of an HTTPS connection to the server that initially set the cookie will be able to read the cookie's value. JavaScript running from a rogue file or email should not be able to successfully read the cookie value to copy into the custom header. Even though the csrf-token cookie may be automatically sent with the rogue request, subject to the cookies SameSite policy, the server will still expect a valid X-Csrf-Token header.
```
Comment (szh) [inline cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md:79]:
```suggestion
The security of this technique is based on the assumption that only JavaScript running on the client side of an HTTPS connection to the server that initially set the cookie will be able to read the cookie's value. JavaScript running from a rogue file or email should not be able to read the cookie value to copy into the custom header. Even though the csrf-token cookie may be automatically sent with the rogue request, subject to the cookies SameSite policy, the server will still expect a valid X-Csrf-Token header.
```

**Review by szh — CHANGES_REQUESTED** [review]
Just some minor nits, otherwise looks great! Thank you!

**Comment by phalgunv** [comment]
Thank you @szh. I have addressed the comments.

**Review by szh — APPROVED** [review]

**Pull request merged**